File: backend/PJATK.TravelAgency/PJATK.TravelAgency.WebApi/Models/Trip.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;

namespace PJATK.TravelAgency.WebApi.Models
{
    public class Trip
    {
        [Key]
        public Guid Id { get; set; }

        [StringLength(255)]
        public string Name { get; set; }

        [StringLength(255)]
        [Required]
        public string City { get; set; }

        [Required]
        public DateTime DepartureDate { get; set; }

        [Required]
        public DateTime ArrivalDate { get; set; }

        [Required]
        public decimal Price { get; set; }

        [StringLength(555)]
        public string Describe { get; set; }

        [StringLength(255)]
        public string DeparturePlace { get; set; }

        [Required]
        public int NumberOfPlaces { get; set; }

        [Required]
        public int AvaiableNumberOfPlaces { get; set; }

        public bool Archive { get; set; }
        public int Promote { get; set; }
               
        //TODO: jak zrobić relację jeden do wielu w EF Code First?
        public HashSet<Reservation> Reservations { get; set; }

        [ForeignKey("Rating")]
        public Guid RatingId { get; set; }

        public Rating Rating { get; set; }
    }
}

File: frontend/src/app/model/trip.ts
export class Trip {
    
    constructor(
        public id: number,
        public name: string, 
        public city: string, 
        public departureDate: Date, 
        public arrivalDate: Date, 
        public price: number,
        public describe: string, 
        public departurePlace: string, 
        public numberOfPlaces: number, 
        public availableNumberOfPlaces: number, 
        public archive: boolean, 
        public promote: number,
        public averageRating: number, 
        public photos: string) {
    }

    static createEmptyTrip(): Trip{
        return new Trip(null,null,null,null,null,null,null,null,null,null,null,null,0,null);
    }

    static createTripFromApiTrip(obj){
        return new Trip(obj.Id, obj.Name, obj.City, obj.DepartureDate, obj.ArrivalDate, obj.Price, obj.Describe, obj.DeparturePlace, obj.NumberOfPlaces, obj.AvailableNumberOfPlaces, obj.Archive, obj.Promote, obj.AverageRating, obj.Photos[0]);
    }
}